src/rpc/blockchain.cpp

```diff
@@ -940,6 +940,106 @@ static UniValue getblock(const JSONRPCRequest& request)
+            "  \"size\" : n,            (numeric) The block size\n"
+            "  \"strippedsize\" : n,    (numeric) The block size excluding witness data\n"
+            "  \"weight\" : n           (numeric) The block weight as defined in BIP 141\n"
+            "  \"height\" : n,          (numeric) The block height or index\n"
+            "  \"version\" : n,         (numeric) The block version\n"
+            "  \"versionHex\" : \"00000000\", (string) The block version formatted in hexadecimal\n"
+            "  \"merkleroot\" : \"xxxx\", (string) The merkle root\n"
+            "  \"tx\" : [               (array of string) The transaction ids\n"
+            "     \"transactionid\"     (string) The transaction id\n"
+            "     ,...\n"
+            "  ],\n"
+            "  \"time\" : ttt,          (numeric) The block time in seconds since epoch (Jan 1 1970 GMT)\n"
+            "  \"mediantime\" : ttt,    (numeric) The median block time in seconds since epoch (Jan 1 1970 GMT)\n"
+            "  \"nonce\" : n,           (numeric) The nonce\n"
+            "  \"bits\" : \"1d00ffff\", (string) The bits\n"
+            "  \"difficulty\" : x.xxx,  (numeric) The difficulty\n"
+            "  \"chainwork\" : \"xxxx\",  (string) Expected number of hashes required to produce the chain up to this block (in hex)\n"
+            "  \"nTx\" : n,             (numeric) The number of transactions in the block.\n"
+            "  \"previousblockhash\" : \"hash\",  (string) The hash of the previous block\n"
+            "  \"nextblockhash\" : \"hash\"       (string) The hash of the next block\n"
+            "}\n"
+                    },
+                    RPCResult{"for verbosity = 2",
+            "{\n"
+            "  ...,                     Same output as verbosity = 1.\n"
+            "  \"tx\" : [               (array of Objects) The transactions in the format of the getrawtransaction RPC. Different from verbosity = 1 \"tx\" result.\n"
+            "         ,...\n"
+            "  ],\n"
+            "  ,...                     Same output as verbosity = 1.\n"
+            "}\n"
+                    },
+                },
+                RPCExamples{
+                    HelpExampleCli("getblockbyheight", "0")
+            + HelpExampleRpc("getblockbyheight", "42")
+                },
+    };
+
+    if (request.fHelp || !help.IsValidNumArgs(request.params.size())) {
+        throw std::runtime_error(help.ToString());
+    }
+
+    int verbosity = 1;
+    if (!request.params[1].isNull()) {
+        if(request.params[1].isNum())
+            verbosity = request.params[1].get_int();
+        else
+            verbosity = request.params[1].get_bool() ? 1 : 0;
+    }
+
+    CBlock block;
+    // Check if block exists or is negative
+    if(::ChainActive().Height() < request.params[0].get_int() || request.params[0].get_int() <= -1)
```

Comment:
Why no check for param[0].isNull()?

Reply:
Good point

Reply:
What do you mean by isNull?
If the param is empty or if the user has selected block zero, which is a valid block

Reply:
`<= -1` should be `== BLOCK_DOESNT_EXIST` and define block doesn't exist to -1. I would also bet this constant is define in a header already. Find it please. We also can't have negative blocks so no need to check for less than -1.

Reply:
> `<= -1` should be `== BLOCK_DOESNT_EXIST` and define block doesn't exist to -1.

Why? This is a user entered value, it could literally be anything, not just `-1`. A user isn't going to search specifically for `-1` just so they can get an error. It could be `-16654` or whatever you want. It is safer to check whether the value is `<= -1` rather than `-1`.

Also, why would this need to be defined as a constant? It really isn't useful as one.

> I would also bet this constant is define in a header already. Find it please. We also can't have negative blocks so no need to check for less than -1.

It probably isn't defined anywhere as there is no need to define it anywhere.

Reply:
> Why?
It's easier to read and maintain if we check for the valid paths.  The documentation says verbose 0-3, uses a weak type (int vs enum), then check for unrelated int values mid code - it's confusing. We can't just arbitrarily pick error cases to check for here. Why not check if the user inputted a string, emoji, a number greater than `INT_MAX` too? 

> It probably isn't defined anywhere as there is no need to define it anywhere.
You're right it's not define but it would be useful.

Reply:
> It's easier to read and maintain if we check for the valid paths. The documentation says verbose 0-3, uses a weak type (int vs enum), then check for unrelated int values mid code - it's confusing.

You're getting your conversations mixed up. This isn't about verbosity.

> We can't just arbitrarily pick error cases to check for here.

We aren't.

> Why not check if the user inputted a string, emoji, a number greater than `INT_MAX` too?

That's why RPCs should use [`RPCTypeCheck`](https://github.com/bitcoin/bitcoin/blob/master/src/rpc/util.cpp#L17). That (and `UniValue` itself) will handle type and integer out of bound errors.